### PR TITLE
Bacthprocessing2

### PR DESCRIPTION
**Pull request by tamajongnc**

### CODE REVIEW

File: antidox/setup.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
+#
```
Comment (dborkan) [inline antidox/setup.py]: Can you explain why this file is needed or how it's used?  It doesn't seem like it's imported by perspective.py

File: antidox/perspective.py [hunk 2]
```diff
@@ -170,71 +158,101 @@ def main(argv):
...
+  parser.add_argument('--temp_location', help='cloud storage path for temp files \
+                                              must begin with gs://')
+  args, pipe_args = parser.parse_known_args(argv)
   apikey_data, perspective, dlp = get_client()
-
-
-if __name__ == '__main__':
-  main(sys.argv[1:])
-  with beam.Pipeline(options=PipelineOptions()) as pipeline:
+  options = PipelineOptions(pipe_args)
```
Comment (dborkan) [inline antidox/perspective.py]: NIT: there are about 9 lines here for setting the options, you might consider moving this to a new function like `def get_pipeline_options()` that returns this object

File: antidox/perspective.py [hunk 3]
```diff
@@ -170,71 +158,101 @@ def main(argv):
...
     if args.csv_file:
-      comments = pipeline | 'ReadMyFile' >> beam.io.ReadFromText(args.csv_file)
+      comments = pipeline | 'ReadMyFile' >> beam.io.ReadFromText(pd.read_csv(args.csv_file))
     if args.sql_query:
       comments = (
           pipeline
           | 'QueryTable' >> beam.io.Read(beam.io.BigQuerySource(
               query=args.sql_query,
-              use_standard_sql=True))
+              use_standard_sql=True)) \
```
Comment (dborkan) [inline antidox/perspective.py]: Is this `\` needed?  If so is there a reason it isn't also needed on line 188 (after `pipeline`)?

File: antidox/perspective.py [hunk 4]
```diff
@@ -170,71 +158,101 @@ def main(argv):
...
+      """class for NDJson"""
+
+      # pylint: disable=fixme, no-self-use
+      # pylint: disable=fixme, inconsistent-return-statements
+      def process(self, element):
+        """Takes toxicity and dlp results and converst them to NDjson"""
+        try:
+          dlp_response = dlp_request(dlp, apikey_data, element)
+          perspective_response = perspective_request(perspective, element)
+          if contains_toxicity(perspective_response):
```
Comment (dborkan) [inline antidox/perspective.py]: With this logic if a comment contains both toxicity and pii, it looks like we will only write JSON with the 'Toxicity' element, and no 'pii_detected' element.  Would it be possible to rewrite this a little to be more like this:
```
data = {
  'comment': element,
  'toxicity': str(perspective_response['attributeScores']['TOXICITY']['summaryScore']['value'])},
  'pii_detected': str(pii_type)
}
```
And then have only 1 return statement?

File: antidox/setup.py (hunk 1 above)
Comment (tamajongnc) [reply antidox/setup.py]: it used used has all the requirements for the distributed systems.

--setup_file ./setup.py

**Pull request merged**